pkg/controllers/updaterun/validation.go

```diff
@@ -258,6 +256,22 @@ func validateDeleteStageStatus(
 		klog.ErrorS(unexpectedErr, "Failed to find the deletionStageStatus in the clusterStagedUpdateRun", "clusterStagedUpdateRun", updateRunRef)
 		return -1, fmt.Errorf("%w: %s", errStagedUpdatedAborted, unexpectedErr.Error())
 	}
+
+	// Validate whether toBeDeletedBindings are a subnet of the clusters in the delete stage status.
+	// We only validate if it's a subnet because clusters can unjoin from the fleet and we allow that.
```

Comment:
the real reason is not about cluster leave the fleet ( we don't allow that for updating ones). It's about the action on the cluster in the deleting stages is to delete the binding so they will disappear in the next round of validation.

Reply:
Yea, I realized that when I write the execution code. Actually the same check is done during the deletionStage execution, so I tend to remove this redundancy here.